### ISSUE TITLE
[Stranglethorn Vale][Loot][Epic][Hyacinth Macaw] Pirate's loot looks wrong

### ISSUE THREAD

**Issue by Orixim** [chromiecraft/chromiecraft#2362]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [X] 40-49
- [ ] 50-59

### Current Behaviour

This might be pure unluck rather than a bug, but at somepoint unluck cannot explain it all.

I have been killing the pirates in both boats in the southeast corner of Stranglethorn Vale for some time now. I can't say for sure how many I've killed, but my estimation is more than 30.000 pirates.

![Screenshot_51](https://user-images.githubusercontent.com/88677774/142745148-aa76affb-33a8-47c4-91f5-d5bc7061a756.png)

I am trying to get a Hyacinth Macaw, whose drop rate is around 1 in 10.000 pirates. However, I got all sorts of loot (even single pirates dropping green and blue items together). Just to give an idea, I've gotten:

* +/- 1.500 gold from vendoring pirates loot
* 15 x [First Mate Hat](https://wowgaming.altervista.org/aowow/?item=2955) = 0.1% drop chance from all the pirates
* 2 x [Bow of Searing Arrows](https://wowgaming.altervista.org/aowow/?item=2825) = 0.03% drop chance and only from [Bloodsail Elder Magus](https://wowgaming.altervista.org/aowow/?npc=1653)
* 1 x [Ardent Custodian](https://wowgaming.altervista.org/aowow/?item=868) = 0.03% drop chance and only from [Bloodsail Elder Magus](https://wowgaming.altervista.org/aowow/?npc=1653)
* 1 x [Boots of Avoidance](https://wowgaming.altervista.org/aowow/?item=14549) = **should not even drop here**
* 0 x [Parrot Cage (Hyacinth Macaw)](https://wowgaming.altervista.org/aowow/?item=8494) = 0.01 % drop chance from all the pirates

That being said and based on the [law of large numbers](https://en.wikipedia.org/wiki/Law_of_large_numbers), the Hyacinth Macaw seems very off. I believe it is dropping, since I've seen it on Auction House, but maybe the drop rate is not as it should be? Or perhaps the random number generator that decides against the drop chance is faulty? I know it can be sheer hard luck, but it is getting harder and harder to accept that.

Anyone else have numbers on from their experiences?

### Expected Blizzlike Behaviour

It should drop 1 parrot for each 10000 pirates killed more or less.

### Source

[Altervista](https://wowgaming.altervista.org/aowow/?item=8494)
[WoWHead Classic](https://classic.wowhead.com/item=8494/parrot-cage-hyacinth-macaw)
[WoWHead](https://www.wowhead.com/item=8494/parrot-cage-hyacinth-macaw)

### Steps to reproduce the problem

1. Kill 10.000 pirates
2. Check if parrot dropped

### Extra Notes

I understand this is hard to accept as a bug, since someone already got it (so it's dropping), but it is also hard to accept 30.000+ is hard luck.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/00dc369cb627d728c520cd597b31e9edf2c75eed

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Shineslippers** [comment]
Database definitely has Hyacinth Macaw in loot tables. From what i can say it has same drop's rules as First Mate Hat. Everytime when you got hat it could be pet. Boots of Avoidance contain in rtl that sharing Bloodsail Deckhand and site doesn't showing this. Sorry, looks like it's your luck.

| entry | name | Item | chance | 
| ---: | --- | ---: | ---: | 
| 1561 | Bloodsail Raider | 8494 | 0,01 | 
| 1562 | Bloodsail Mage | 8494 | 0,01 | 
| 1563 | Bloodsail Swashbuckler | 8494 | 0,01 | 
| 1564 | Bloodsail Warlock | 8494 | 0,01 | 
| 1565 | Bloodsail Sea Dog | 8494 | 0,01 | 
| 1653 | Bloodsail Elder Magus | 8494 | 0,01 | 
| 2545 | "Pretty Boy" Duncan | 8494 | 0,02 | 
| 2546 | Fleet Master Firallon | 8494 | 0,01 | 
| 4505 | Bloodsail Deckhand | 8494 | 0,01 | 
| 4506 | Bloodsail Swabby | 8494 | 0,02 |

**Comment by Azcobu** [comment]
I've had a look at the various loot tables for NPCs dropping this item, looking for incorrectly set flags or incorrect loot chance totals or anything like that, but can't see any evidence of an issue. As logged world chat and AH listings both show people are getting it to drop for them, the drop also appears to be working. The chance of not getting the drop after 30000 kills is 0.9999<sup>30000</sup> ~= 5%, so you've been pretty unlucky, but not extraordinarily so.

Will close this issue for lack of evidence for an issue, and hope you have better luck from now on.

**Comment by Orixim** [comment]
So I've finally got it.

![Screenshot_1](https://user-images.githubusercontent.com/88677774/143606052-01376100-5aab-4376-bb64-62b97c45186e.png)

I had to kill about 7k more pirates, I've dropped 1 x [Gloves of Holy Might](https://wowgaming.altervista.org/aowow/?item=867) (0.03% drop chance and only from Bloodsail Elder Magus) and then the parrot.

I've decided to kill them while wearing the First Mate Hat, I have no clue if there is any relationship between them.

So I needed to get about 98% chance (0.9999^37000) to have one dropped for it to drop. I have to say that it feels odd.

Anyway, I just wanted to leave the numbers here and maybe others can do the same to have more data about it. The ticket can remain close I guess.
